### ISSUE TITLE
What does the build method do?

### ISSUE THREAD

**Issue by gaorlov** [JsonApiClient/json_api_client#289]
What does the `Query::Builder#build` method do?

It doesn't behave the same way as `new` and builds the `attributes` hash in a completely different way

```ruby
Example::Client::User.where( id: 68 ).where( name: "Greg" ).build
# => <Example::Client::User:@attributes={"type"=>"users", "filter"=>{"id"=>68, "name"=>"Greg"}}>

Example::Client::User.new id: 68, name: "Greg"
# => <Example::Client::User:@attributes={"type"=>"users", "id"=>68, "name"=>"Greg"}>
```

There's no tests for it and it doesn't seem to be used anywhere. What am I missing?

**Comment by senid231** [comment]
@gaorlov I don't think `#build` method should add any of it's params to attributes on created model

**Comment by senid231** [comment]
fixed/ensured by #326
